infer_unet_process.py

```diff
@@ -0,0 +1,174 @@
+        pass
+
+    def getParamMap(self):
+        # Send parameters values to Ikomia application
+        # Create the specific dict structure (string container)
+        param_map = core.ParamMap()
+        param_map["modelFile"] = self.modelFile
+        param_map["img_scale"] = self.img_scale
+        param_map["num_channels"] = self.num_channels
+        param_map["num_classes"] = self.num_classes
+        param_map["outputFolder"] = self.outputFolder
+        return param_map
+
+
+# --------------------
+# - Class which implements the process
+# - Inherits PyCore.CWorkflowTask or derived from Ikomia API
+# --------------------
+class InferUnet(dataprocess.C2dImageTask):
+
+    def __init__(self, name, param):
+        dataprocess.C2dImageTask.__init__(self, name)
+
+        # Create parameters class
+        if param is None:
+            self.setParam(InferUnetParam())
+        else:
+            self.setParam(copy.deepcopy(param))
+
+    def getProgressSteps(self):
+        # Function returning the number of progress steps for this process
+        # This is handled by the main progress bar of Ikomia application
+        return 1
+
+    def run(self):
+        # Core function of your process
+        # Call beginTaskRun for initialization
+        self.beginTaskRun()
+
+        # Get input :
+        img_input = self.getInput(0)
+        input_image = img_input.getImage()
+
+        # Get parameters :
+        param = self.getParam()
+
+        # load model file after training unet model or use the Carnava pretrained model to test the model
+        path = param.modelFile
+        if os.path.isfile(path):
+            net = UNet(param.num_channels, param.num_classes)
+            net.load_state_dict(torch.load(path))
+        # else use the carnava pretrained model
+        else:
+            net = torch.hub.load('milesial/Pytorch-UNet', 'unet_carvana', pretrained=True, scale=param.img_scale)
+
+        device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+        net.to(device=device)
+        net.eval()
+
+        # segment image
+        # mask prédit par le modèle (probabilités de chaque classe)
+        img = Image.fromarray(input_image)
+        mask = predict_mask(net=net,
+                            full_img=img,
+                            scale_factor=param.img_scale,
+                            device=device)
+
+        # convert prediction to image
+        mask_image = mask_to_image(mask)
+        Pil_img = Image.fromarray(mask_image.astype(np.uint8))
+        # save segmented image
+        # current datetime is used as folder name
+        str_datetime = datetime.now().strftime("%d-%m-%YT%Hh%Mm%Ss")
+        out_img_path = os.path.join(param.outputFolder, str_datetime+'.png')
+        Pil_img.save(out_img_path)
+
+        # Get output :
+        output = self.getOutput(0)
+
+        # Set image of output (numpy array):
+        output.setImage(mask_image)
```

Comment:
Use self.forwardInputImage() to dispay original image as output. You need also to fill a CSemanticSegIO output: all our semantic segmentation algrithms should have the same behavior. You can have a look a infer_detectron2_deeplabv3plus for an example.